File: Java/0082_remove_duplicates_from_sorted_list_ii/0082_remove_duplicates_from_sorted_list_ii.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null || head.next == null)
            return head;
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode current = head;
        ListNode prev = dummy;
        
        while (current != null && current.next != null) {
            int count = 1;
            while (current.next != null && current.val == current.next.val) {
                current = current.next;
                count++;
            }
            if (count == 1) {
                prev = current;
                current = current.next;
            }
            else {
                current = current.next;
                prev.next = current;
            }
        }
        return dummy.next;
    }
}

File: Python/0082_remove_duplicates_from_sorted_list_ii/0082_remove_duplicates_from_sorted_list_ii.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head == None or head.next == None:
            return head
        dummy = ListNode(0)
        dummy.next = head
        current = head
        prev = dummy
        
        while current != None and current.next != None:      
            count = 1
            while current.next != None and current.val == current.next.val:
                current = current.next
                count += 1
            if count == 1:
                prev = current
                current = current.next
            else:
                current = current.next
                prev.next = current
        return dummy.next
